File: users/saveenr/GetLineInformation.py

from datetime import datetime

tstart = datetime.now()

line_count = 0
max_line = ""
max_line_size_bytes = 0
max_line_size_chars = 0
max_line_index = 0
with open( r"E:\Mozilla-Bugzilla-Public-02-December-2013.sql\Mozilla-Bugzilla-Public-02-December-2013.sql") as f:
    for line in f:
		if (len(line)>max_line_size_bytes) :
			max_line = line
			max_line_size_bytes = len(line)
			max_line_index = line_count
			u = unicode(line, "utf-8")
			max_line_size_chars = len(u)
			
		line_count = line_count + 1
tend = datetime.now()
rduration = tend - tstart

print "Line count", line_count
print "Max line length in chars", max_line_size_chars
print "Max line length in bytes", max_line_size_bytes
print "Max line index", max_line_index
print "Total time", rduration.seconds

File: users/saveenr/GetLineInformation.cs
namespace GetLineInformation
{
    class Program
    {
        static void Main(string[] args)
        {
            var start = System.DateTime.Now;

            int line_with_max = 0;
            int max_chars = 0;
            int num_lines = 0;
            string max_line =string.Empty;
            int max_bytes = 0;
            string line;
            string filename = args[0];
            using (var file = new System.IO.StreamReader(filename))
            {
                while ((line = file.ReadLine()) != null)
                {
                    if (line.Length > max_chars)
                    {
                        max_chars = line.Length;
                        max_line = line;
                        max_bytes = System.Text.Encoding.UTF8.GetBytes(max_line).Length;
                        line_with_max = num_lines;
                    }
                    num_lines++;
                }

                file.Close();
            }

            var end = System.DateTime.Now;
            var duration = end - start;

            System.Console.WriteLine("Total time in seconds: {0}", duration.TotalSeconds);
            System.Console.WriteLine("Total lines: {0}", num_lines);
            System.Console.WriteLine("Max chars in line: {0}", max_chars);
            System.Console.WriteLine("Max bytes in line: {0}", max_bytes);
            System.Console.WriteLine("line number: {0}", line_with_max);
        }
    }
}
